mne/channels/montage.py

```diff
@@ -831,14 +862,22 @@ def _backcompat_value(pos, ref_pos):
             # in the old dig
             if ref_dig_point in old_dig:
                 digpoints.append(ref_dig_point)
+        # XXX info['dig'] modified in place
         info['dig'] = _format_dig_points(digpoints, enforce_order=True)
 
         if mnt_head.dev_head_t is not None:
+            # XXX info['dev_head_t'] modified in place
             info['dev_head_t'] = Transform('meg', 'head', mnt_head.dev_head_t)
 
+        fnirs_picks = _picks_to_idx(info, 'fnirs', allow_empty=True)
+        if len(fnirs_picks) > 0:
+            info = _set_montage_fnirs(info, montage)
```

Comment:
To avoid the pain of the next person errantly using `montage` instead of `mnt_head`, after conversion above we could `del montage` so that it's no longer in the namespace as a strong hint that the montage-in-head should be used by the subsequent code.